src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -180,12 +180,12 @@ impl<'a, 'b> ReadOnlyChecker<'a, 'b> {
             | Keccak256 | Equals | If | Sha512 | Sha512Trunc256 | Secp256k1Recover
             | Secp256k1Verify | ConsSome | ConsOkay | ConsError | DefaultTo | UnwrapRet
             | UnwrapErrRet | IsOkay | IsNone | Asserts | Unwrap | UnwrapErr | Match | IsErr
-            | IsSome | TryRet | ToUInt | ToInt | Append | Concat | AsMaxLen | ContractOf
-            | PrincipalOf | ListCons | GetBlockInfo | TupleGet | TupleMerge | Len | Print
-            | AsContract | Begin | FetchVar | GetStxBalance | StxGetAccount | GetTokenBalance
-            | GetAssetOwner | GetTokenSupply | ElementAt | IndexOf => {
-                self.check_all_read_only(args)
-            }
+            | IsSome | TryRet | ToUInt | ToInt | BuffToIntLe | BuffToUIntLe | BuffToIntBe
+            | BuffToUIntBe | StringToInt | StringToUInt | IntToAscii | IntToUtf8 | Append
```

Comment:
Do the new functions use 2 or 1 argument? It looks like this branch performs a "2" argument check, rather than a 1 argument check.